src/lib/arch/riscv/Instruction.cc

```diff
@@ -134,12 +135,16 @@ int64_t Instruction::getKnownOffset() const { return knownOffset_; }
 uint16_t Instruction::getGroup() const {
   uint16_t base = InstructionGroups::INT;
 
+  if (isFloat()) {
+    base = InstructionGroups::FLOAT;
+  }
+
   if (isBranch()) return InstructionGroups::BRANCH;
   if (isLoad()) return base + 8;
   if (isStoreAddress()) return base + 9;
   if (isDivide_) return base + 7;
   if (isMultiply_) return base + 6;
-  if (isShift_) return base + 5;
+  if (isShift_ || isConvert_) return base + 5;
```

Comment:
Is there a reason that `isShift` and `isConvert` share a base?

Reply:
I could split these and move the isConvert_ check into the ```if(isFloat())``` clause above if that makes if more readable

Reply:
I'm a bit confused at your explanation, but I think splitting it so that `isShift` and `isConvert` have differet offsets (i.e. how in AArch64 all types have their own offset). This is mainly for the fact that in future instructions _could_ be added which require `isShift` and `isConvert` to be seperate...

Reply:
After offline discussion this will be left as is, but another PR should be created to clean up InstructionGroup logic

Reply:
See issue #347